spec/simplecov_spec.rb

```diff
@@ -107,5 +107,89 @@
         end
       end
     end
+
+    describe ".set_exit_exception" do
+      context "when an exception has occurred" do
+        let(:error) { StandardError.new "SomeError" }
+
+        after do
+          # Clear the exit_exception
+          SimpleCov.set_exit_exception
+        end
+
+        it "captures the current exception" do
+          begin
+            raise error
+          rescue
+            SimpleCov.set_exit_exception
+            expect(SimpleCov.exit_exception).to be(error)
+          end
+        end
+      end
+
+      context "when an exception has not occurred" do
+        it "does set the exit_exception" do
+          SimpleCov.set_exit_exception
+          expect(SimpleCov.exit_exception).to eq(nil)
+        end
+      end
+    end
+
+    describe ".exit_exception" do
+      let(:error) { StandardError.new "SomeError" }
+
+      before do
+        begin
+          raise error
+        rescue
+          SimpleCov.set_exit_exception
+        end
+      end
+
+      after do
+        # Clear the exit_exception
+        SimpleCov.set_exit_exception
+      end
+
+      it "returns the exit exception" do
+        expect(SimpleCov.exit_exception).to be(error)
```

Comment:
imo we already tested this basically in the tests before this, so I'd be good without this spec :)